hooks/hiero_get_shot.py

```diff
@@ -20,14 +20,22 @@ def execute(self, task, item, data, **kwargs):
         Takes a hiero.core.TrackItem as input and returns a data dictionary for
         the shot to update the cut info for.
         """
-        # get the parent sequence for the Shot
-        sequence = self._get_sequence(item, data)
+
+        # get the parent entity for the Shot
+        parent = self.parent.execute_hook(
+            "hook_get_shot_parent",
+            hiero_sequence=item.parentSequence(),
+            data=data
+        )
```

Comment:
@manneohrstrom and I had a chat. The take away was that we should definitely not break anyone that has overridden the existing `hook_get_shot`. We also can't make any assumptions about the contents of that hook other than the expected value returned from the `execute` method. The new code `Cut` support needs a parent entity for the shot, but we don't want to assume this is a `Sequence`. The compromise we discussed was for the new implementation of the hook to include 2 methods: `get_shot` and `get_shot_parent` (or something similarly named). We will attempt to call `get_shot` as needed and if an exception is raised (we may need a new exception in core for [this specific case](https://github.com/shotgunsoftware/tk-core/blob/master/python/tank/hook.py#L278) we will fall back to trying `execute`. When we only need the parent, we'll attempt to call `get_shot_parent` and if it doesn't exist then the `Cut` will be created without a parent. This implies that nothing will break, but we'll need to document the change such that clients who upgrade and have overridden the existing hook will need to know to shuffle their code a bit to get the proper parenting structure for the new `Cut` schema. 

@manneohrstrom, please let me know if I forgot something or if there's need to clarify!
@robblau what do you think?